src/server/cron/taskUtil.js

```diff
@@ -2,6 +2,9 @@
 export const DISPOSE_ENROLLMENTS_TASK = 'verification/dispose_enrollments'
 
 export const scheduleDisposalTask = async (storage, enrollmentIdentifier, executeAt): Promise<DelayedTaskRecord> => {
-  await storage.cancelTasksQueued(DISPOSE_ENROLLMENTS_TASK, { 'subject.enrollmentIdentifier': enrollmentIdentifier })
+  await storage.cancelTasksQueued(DISPOSE_ENROLLMENTS_TASK, {
+    'subject.enrollmentIdentifier': enrollmentIdentifier,
+    'subject.executeAt': executeAt
```

Comment:
```suggestion
```

no, we actually need to remove **any** task for that enrollmentIdentifier before add the new one. so we have to check  'subject.enrollmentIdentifier' only in the filters